File: binarysearch/A_Binary_Search.cpp
#include <iostream>
using namespace std;

int binarySearch(int[] arr, int target) {
    int left = 0;
    int right = arr.length - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    // target not found in array
    return -1;
}


int main(){
    int n, k;
    cin>>n>>k;
    vector<int> a(n);
    for (int i; i<n; i++){
        cin>>a[i];
    }
    
    for (int i; i<k; i++){
        int x;
        cin>>x;

        int l = 0;
        int h = n-1;
        bool flag = false;

        while (l<=h){
            int m = l-(h-l)/2;
            if (a[m]==x){
                flag = true;
                return;
            }else if(a[m]>x){
                h = m-1;
            }
            else{
                l = m+1;
            }
        }
        if(flag){
            cout<<"YES";
        }
        else{
            cout<<"NO";
        }
    }

}

File: binarysearch/A_Binary_Search.py
n, k = list(map(int, input().split()))

a = list(map(int, input().split()))
a.sort()
queries = list(map(int, input().split()))

for query in queries:
    low = 0
    high = len(a)-1
    flag = "NO"

    while low<=high:
        mid = low+(high-low)//2

        if a[mid] == query:
            flag = "YES"
            break
        elif a[mid]>query:
            high = mid-1
        else:
            low = mid+1
    
    print(flag)